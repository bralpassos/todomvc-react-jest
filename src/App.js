import React from 'react';
import TodoItem from './TodoItem.js'

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentItem: '',
      items : []
    }
  }

  handleChange = (e) => {
    this.setState({ currentItem: e.target.value });
  }

  addNewItem = (e) => {
    const tempItem = this.state.items;
    const newItem = this.state.currentItem;
    const hasItem = tempItem.indexOf(newItem) >= 0;

    console.log(hasItem);

    if (e.key === 'Enter' && !hasItem) {
      tempItem.push(newItem);
      this.setState({ 
        items: tempItem,
        currentItem: ''
      });
    }
  } 

  render () {
    return (
      <div className="App">
        <header className="header">
          <h1>todos</h1>
          <input
            className="new-todo"
            placeholder="What needs to be done?"
            value={this.state.currentItem}
            onChange={this.handleChange}
            onKeyPress={this.addNewItem}
          />
        </header>
        <TodoItem items={this.state.items} />
      </div>
    );
  }
}

export default App;