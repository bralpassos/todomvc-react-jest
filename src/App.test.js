import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Verificar se ao renderizar, renderiza uma lista vazia', () => {
  const component = renderer.create(<App />);
  expect(component).toMatchSnapshot();
});

it('Não deve adicionar se não apertar Enter', () => {
  const { getByPlaceholderText } = render(<App />);

  expect(getByPlaceholderText("What needs to be done?")).toBeTruthy();
});