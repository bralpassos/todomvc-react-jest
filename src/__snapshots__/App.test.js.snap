// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verificar se ao renderizar, renderiza uma lista vazia 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      className="new-todo"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="What needs to be done?"
      value=""
    />
  </header>
  <section
    className="main"
  >
    <ul
      className="todo-list"
    >
      
    </ul>
  </section>
</div>
`;
